src/CharsetDetector.cs

```diff
@@ -371,43 +371,39 @@ private void FindInputState(byte[] buf, int len)
 
         private static string FindCharSetByBom(byte[] buf, int len)
         {
-            string bomSet = null;
-            if (len > 3)
+            if (len < 2 || buf.Length < 2)
+                return null;
+
+            var buf0 = buf[0];
+            var buf1 = buf[1];
+            
+            if (buf0 == 0xEF && buf1 == 0xBB && buf.Length > 2 && len > 2 && buf[2] == 0xBF)
+                return CodepageName.UTF8;
+
+            if (buf0 == 0xFE && buf1 == 0xFF)
             {
-                //todo UTF bom of only 3 bytes isn't recognized
-                var b1 = buf[0];
-                var b2 = buf[1];
-                var b3 = buf[2];
-                var b4 = buf[3];
-                switch (b1)
-                {
-                    case 0xEF:
-                        if (0xBB == b2 && 0xBF == b3)
-                            bomSet = CodepageName.UTF8;
-                        break;
-                    case 0xFE:
-                        if (0xFF == b2 && 0x00 == b3 && 0x00 == b4)
-                            // FE FF 00 00  UCS-4, unusual octet order BOM (3412)
-                            bomSet = CodepageName.X_ISO_10646_UCS_4_3412;
-                        else if (0xFF == b2)
-                            bomSet = CodepageName.UTF16_BE;
-                        break;
-                    case 0x00:
-                        if (0x00 == b2 && 0xFE == b3 && 0xFF == b4)
-                            bomSet = CodepageName.UTF32_BE;
-                        else if (0x00 == b2 && 0xFF == b3 && 0xFE == b4)
-                            // 00 00 FF FE  UCS-4, unusual octet order BOM (2143)
-                            bomSet = CodepageName.X_ISO_10646_UCS_4_2143;
-                        break;
-                    case 0xFF:
-                        if (0xFE == b2 && 0x00 == b3 && 0x00 == b4)
-                            bomSet = CodepageName.UTF32_LE;
-                        else if (0xFE == b2)
-                            bomSet = CodepageName.UTF16_LE;
-                        break;
-                } // switch
+                // FE FF 00 00  UCS-4, unusual octet order BOM (3412)

```

Comment:
I really prefer some unit tests for these. If needed, you could make this method `internal`

Reply:
They already exist ;)

<details>

https://github.com/CharsetDetector/UTF-unknown/blob/8b33ce8fe9de18ea4d02cef59a5455c0dfcc50ef/tests/CharsetDetectorTest.cs#L77-L80

https://github.com/CharsetDetector/UTF-unknown/blob/8b33ce8fe9de18ea4d02cef59a5455c0dfcc50ef/tests/CharsetDetectorTest.cs#L86-L89

https://github.com/CharsetDetector/UTF-unknown/blob/8b33ce8fe9de18ea4d02cef59a5455c0dfcc50ef/tests/CharsetDetectorTest.cs#L96-L100

https://github.com/CharsetDetector/UTF-unknown/blob/8b33ce8fe9de18ea4d02cef59a5455c0dfcc50ef/tests/CharsetDetectorTest.cs#L107-L111

https://github.com/CharsetDetector/UTF-unknown/blob/8b33ce8fe9de18ea4d02cef59a5455c0dfcc50ef/tests/CharsetDetectorTest.cs#L118-L121

https://github.com/CharsetDetector/UTF-unknown/blob/8b33ce8fe9de18ea4d02cef59a5455c0dfcc50ef/tests/CharsetDetectorTest.cs#L127-L130

https://github.com/CharsetDetector/UTF-unknown/blob/8b33ce8fe9de18ea4d02cef59a5455c0dfcc50ef/tests/CharsetDetectorTest.cs#L136-L139

</details>

Reply:
Why do we need this code then if no tests has changed? ;)

Reply:
This isn't a refactor, isn't?

Reply:
Ahh. Wait a second!

Reply:
Added tests